packages/flutter/lib/src/material/button_style.dart

```diff
@@ -210,6 +210,25 @@ class ButtonStyle with Diagnosticable {
   final MaterialStateProperty<double?>? elevation;
 
   /// The padding between the button's boundary and its child.
+  ///
+  /// The vertical padding is adjusted based on [visualDensity].
```

Comment:
The padding is first resolved (the one specified if set, otherwhise the default one) then changed based on visual density.
Here is where this logic is located:
https://github.com/flutter/flutter/blob/4c46030927626782555918be39985ff95de83438/packages/flutter/lib/src/material/button_style_button.dart#L396-L400

Reply:
Thanks for clarifying that. Since this is somewhat unusual, it might be good to emphasize it a little. For example: 
```
The vertical aspect of the default or user-specified padding is adjusted automatically based on visualDensity.
```

Reply:
Thanks a lot for this great rewording and for all the other suggestions (today I learned a new english word: **wordsmithing** :tada:)